From: Kalesh AP <kalesh.purayil@emulex.com>
Date: Tue, 20 Jan 2015 03:51:47 -0500
Subject: [PATCH 101/145] be2net: issue function reset cmd in resume path
Patch-mainline: v4.0-rc1
Git-commit: 9a6d73d9f297fd1bf7514bf2b1dc2f4b48c638b9
References: bsc#921784 FATE#318561

The Lancer FW is picky about requiring a function reset FW cmd as a part
of the initialization sequence.

Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/emulex/benet/be_main.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -5051,6 +5051,10 @@ static int be_resume(struct pci_dev *pde
 	if (status)
 		return status;
 
+	status = be_cmd_reset_function(adapter);
+	if (status)
+		return status;
+
 	be_intr_set(adapter, true);
 	/* tell fw we're ready to fire cmds */
 	status = be_cmd_fw_init(adapter);
